utility_analysis_new/histograms.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2022 OpenMined.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pipeline_dp
+from pipeline_dp import pipeline_backend
+from dataclasses import dataclass
+import operator
+from typing import List
+
+
+@dataclass
+class FrequencyBin:
+    """Represents 1 bin of the histogram.
+
+  The bin represents integers between 'lower' (inclusive) and 'upper'
+   (exclusive, not stored in this class, but uniquely determined by 'lower').
+
+  Attributes:
+      lower: the lower bound of the bin.
+      count: the number of elements in the bin.
+      sum: the sum of elements in the bin.
+      max: the maximum element in the bin (which might be smaller than upper-1).
```

Comment:
nit: would it make sense to say "smaller or equal to the upper-1" or not?

Reply:
yes, it makes sense.